src/xrEngine/Environment.cpp

```diff
@@ -550,6 +550,9 @@ void CEnvironment::calculate_dynamic_sun_dir()
 
     clamp(cosAZ, -1.0f, 1.0f);
     float AZ = acosf(cosAZ);
+    extern u32 ps_r_sun_direction;
```

Comment:
This should be changed into a member of `CEnvironment` called `m_sun_direction`. Then you can set it by making a command class that sets `g_pGamePersistent->Environment().m_sun_direction` in `Execute()`.

See https://github.com/OpenXRay/xray-16/blob/dev/src/xrEngine/xr_ioc_cmd.cpp#L463 as an example. Also you could just allow arbitrary sun directions if you make it a float in degrees and convert it to radians when setting `m_sun_direction`.